File: 은지/LV2/오픈채팅방.py
def solution(record):
    answer = []
    info = {}
    for v in record:
        x = v.split()
        if x[0] != 'Leave':
            info[x[1]] = x[2]
    for v in record:
        x = v.split()
        if x[0] == 'Enter':
            message = info[x[1]] + "님이 들어왔습니다."
            answer.append(message)
        elif x[0] == "Leave":
            message = info[x[1]] + "님이 나갔습니다."
            answer.append(message)
    return answer

File: 은지/LV2/오픈채팅방.js
function solution(record) {
  var answer = [];
  var info = {}
  for (let v of record) {
    const x = v.split(" ")
    if (x[0] !== 'Leave') {
      info[x[1]] = x[2]
    }
  }
  for (let w of record) {
    const y = w.split(" ")
    var message = ''
    if (y[0] === 'Enter') {
      message = info[y[1]] + "님이 들어왔습니다."
      answer.push(message)
    }
    else if (y[0] === 'Leave') {
      message = info[y[1]] + "님이 나갔습니다."
      answer.push(message)
    }
  }
  return answer;
}